pkgs/utils/src/log.zig

```diff
@@ -0,0 +1,106 @@
+const std = @import("std");
+const builtin = @import("builtin");
+
+// having activeLevel non comptime and dynamic allows us env based logging and even a keystroke activated one
+// on a running client, may be can be revised later
+pub fn log(comptime scope: @Type(.enum_literal), activeLevel: std.log.Level, comptime level: std.log.Level, comptime fmt: []const u8, args: anytype) void {
+    if (@intFromEnum(level) > @intFromEnum(activeLevel)) {
+        return;
+    }
+
+    const system_prefix = if (builtin.target.os.tag == .freestanding) "zkvm" else "zeam";
+
+    const scope_prefix = "(" ++ switch (scope) {
+        std.log.default_log_scope => system_prefix,
+        else => system_prefix ++ "-" ++ @tagName(scope),
+    } ++ "): ";
+    const prefix = "[" ++ comptime level.asText() ++ "] " ++ scope_prefix;
+
+    if (builtin.target.os.tag == .freestanding) {
+        const io = @import("zkvm").io;
+        var buf: [1024]u8 = undefined;
```

Comment:
It'd be better to use an allocator here, since some zkvms don't have a very large stack. You could do have a singleton logger that takes the allocator in its `init` funciton.